server/src/main/java/org/apache/iotdb/db/writelog/io/SingleFileLogReader.java

```diff
@@ -53,54 +54,69 @@
 
   private boolean fileCorrupted = false;
 
+  private boolean hasNext = false;
+
   public SingleFileLogReader(File logFile) throws FileNotFoundException {
     open(logFile);
   }
 
   @Override
   public boolean hasNext() {
-    try {
-      if (batchLogReader != null && batchLogReader.hasNext()) {
-        return true;
-      }
-
-      if (logStream.available() < LEAST_LOG_SIZE) {
-        return false;
-      }
 
-      int logSize = logStream.readInt();
-      if (logSize <= 0) {
-        return false;
-      }
-      buffer = new byte[logSize];
+    if (Objects.nonNull(batchLogReader)) {
+      hasNext = batchLogReader.hasNext();
 
-      int readLen = logStream.read(buffer, 0, logSize);
-      if (readLen < logSize) {
-        throw new IOException("Reach eof");
+      if (hasNext) {
+        return hasNext;
       }
+    }
 
-      final long checkSum = logStream.readLong();
-      checkSummer.reset();
-      checkSummer.update(buffer, 0, logSize);
-      if (checkSummer.getValue() != checkSum) {
-        throw new IOException(String.format("The check sum of the No.%d log batch is incorrect! In "
-            + "file: "
-            + "%d Calculated: %d.", idx, checkSum, checkSummer.getValue()));
+    try {
+      while (logStream.available() > LEAST_LOG_SIZE) {
```

Comment:
why add `while` here?

Reply:
if `batchLogReader ` is null, then we should find a position in logStream where we create BatchLogReader and make `hasNext()` is true repeatedly, it's in fact a loop

Reply:
I wonder how could we get a new BatchLogReader but it has nothing to read and yet it is not corrupted. If there is indeed such a situation, please add associated unit test.